Show/Hide events table columns

hriday-panchasara opened the pull request:
- Added an optional 'Options menu button' in AbstractOutputComponent's Title Bar:
<img width="137" alt="Screen Shot 2022-04-14 at 3 47 48 PM" src="https://user-images.githubusercontent.com/92893187/163472738-d5629849-005c-4b40-bd3f-078441fc518e.png">

- Events table implements this button to open a dropdown list that allows users to toggle visibility of multiple columns
<img width="543" alt="Screen Shot 2022-04-14 at 3 47 14 PM" src="https://user-images.githubusercontent.com/92893187/163472666-39595f2e-50a3-4ae5-b686-ef39c5e5487d.png">

PatrickTasse commented on packages/react-components/style/output-components-style.css:
```diff
@@ -248,3 +253,21 @@ canvas {
 }
+
+.options-menu-drop-down {
+    position: absolute;
```
It seems to me that as a menu popup, the dialog should appear in a position relative to the mouse cursor or the menu icon.

Especially that where it is placed now, and with no user indication of how to close the dialog (click outside?), the user might mistakenly think that the [x] close button is to close the dialog, but it closes the table output.

PatrickTasse commented on packages/react-components/src/components/abstract-output-component.tsx:
```diff
@@ -166,7 +166,7 @@ export abstract class AbstractOutputComponent<P extends AbstractOutputProps, S e
 
     private closeOptionsMenu(event: Event): void {
-        if (event.currentTarget instanceof Node && this.optionsMenuRef.current?.contains(event.currentTarget)) {
+        if (event.target instanceof Node && this.optionsMenuRef.current?.contains(event.target)) {
```
This change belongs in the first commit?

PatrickTasse commented on packages/react-components/style/output-components-style.css:
```diff
@@ -257,15 +257,15 @@ canvas {
     position: absolute;
     left: 100%;
-    width: 300px;
+    width: 250px;
```
These changes belong in the first commit?

PatrickTasse commented on packages/react-components/src/components/abstract-output-component.tsx:
```diff
@@ -94,7 +100,11 @@ export abstract class AbstractOutputComponent<P extends AbstractOutputProps, S e
                 <i id={this.props.traceId + this.props.outputDescriptor.id + 'handleSpinner'} className='fa fa-refresh fa-spin'
                     style={{ marginTop: '5px', visibility: 'hidden'}} />
+                {this.state.optionsDropdownOpen !== undefined && <button title='Show View Options' className='remove-component-button' onClick={this.openOptionsMenu}>
+                    <FontAwesomeIcon icon={faBars} />
```
I think the options menu button should be snapped below the close button, not floating when output height is resized.

PatrickTasse commented on packages/react-components/src/components/table-output-component.tsx:
```diff
@@ -653,4 +654,56 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+                        <th>Show/Hide</th>
+                        <th>Column Name</th>
+                    </tr>
+                </thead>
```
In my opinion, this table header is unnecessary and I would prefer the checkboxes much closer to the column names.

hriday-panchasara commented:
@PatrickTasse Changes Made:
- Rebased so all changes belong to one commit
- Removed table header
- Options menu snapped below the close button. Clicking on this would open a dropdown menu. Dropdown menu has 'Toggle Columns' option - when clicked the table will appear. The idea is to have this dropdown also include 'Export View' and 'Pin View' options. I'll add them to this dropdown once this item has merged and I'll have to resolve merge conflicts in those PR's
![Screen Shot 2022-04-22 at 4 25 25 PM](https://user-images.githubusercontent.com/92893187/164797387-ad60cec6-95aa-4ac5-b087-e913e953f68d.png)
- Dropdown appears relative to menu icon

bhufmann reviewed (approved):
Looks good to me. Thanks for this feature.
